zerver/worker/queue_processors.py

```diff
@@ -587,24 +587,23 @@ def consume(self, event: Dict[str, Any]) -> None:
             user_profile_id: int = event["user_profile_id"]
             batch_duration = datetime.timedelta(seconds=self.BATCH_DURATION)
 
-            with transaction.atomic():
```

Comment:
I think we want `transaction.atomic`; we just need to be catching `IntegrityError` when thrown by the relevant block.